pydoctor/epydoc/markup/_types.py

```diff
@@ -0,0 +1,188 @@
+                
+                # second level
+                if isinstance(super_parent, nodes.document):
+                    # only text in paragraph nodes are taken into account
+                    if isinstance(node, nodes.Text):
+                        # Tokenize the Text node with the same method TypeDocstring uses.
+                        self.tokens.extend(TypeDocstring._tokenize_type_spec(node.astext()))
+                    else:
+                        self.tokens.append(node)
+                        raise nodes.SkipNode()
+    
+        tokenizer = Tokenizer(spec)
+        spec.walk(tokenizer)
+        self.warnings.extend(tokenizer.warnings)
+        return tokenizer.tokens
+
+    def _convert_obj_tokens_to_stan(self, tokens: List[Tuple[Union[str, nodes.Node], TokenType]], 
+                                    docstring_linker: DocstringLinker) -> List[Tuple[Union[str, Tag, nodes.Node], TokenType]]:
+        """
+        Convert L{TokenType.OBJ} and PEP 484 like L{TokenType.DELIMITER} type to stan, merge them together. Leave the rest untouched. 
+
+        Exemple:
+
+        >>> tokens = [("list", TokenType.OBJ), ("(", TokenType.DELIMITER), ("int", TokenType.OBJ), (")", TokenType.DELIMITER)]
+        >>> ann._convert_obj_tokens_to_stan(tokens, NotFoundLinker())
+        ... [(Tag('code', children=['list', '(', 'int', ')']), TokenType.OBJ)]
+        
+        @param tokens: List of tuples: C{(token, type)}
+        """
+
+        combined_tokens: List[Tuple[Union[str, Tag], TokenType]] = []
+
+        open_parenthesis = 0
+        open_square_braces = 0
+
+        for _token, _type in tokens:
+
+            if _type is TokenType.OBJ:
+                new_token = docstring_linker.link_xref(_token, _token, self._lineno)
+                if open_square_braces + open_parenthesis > 0:
+                    try: last_processed_token = combined_tokens[-1]
+                    except IndexError: 
+                        # weird
+                        combined_tokens.append((_token, _type))
+                    else:
+                        if last_processed_token[1] is TokenType.OBJ and isinstance(last_processed_token[0], Tag):
+                            # Merge with last Tag
+                            last_processed_token[0](*new_token.children)
+                        else:
+                            # weird
+                            combined_tokens.append((new_token, _type))
+                else:
+                    combined_tokens.append((new_token, _type))
+
+            elif _type is TokenType.DELIMITER: 
+                if _token == "[": open_square_braces += 1
+                elif _token == "(": open_parenthesis += 1
+
+                if open_square_braces + open_parenthesis > 0:
+                    try: last_processed_token = combined_tokens[-1]
+                    except IndexError: 
+                        # weird
+                        combined_tokens.append((_token, _type))
```

Comment:
Not even sure how we could test this block honestly. Maybe a case to use `assert`  again.